src/wrapper/jnienv.rs

```diff
@@ -1996,83 +1996,83 @@ impl<'a> JNIEnv<'a> {
     /// [`get_double_array_elements`](struct.JNIEnv.html#method.get_double_array_elements)
     /// And the associated [`AutoArray`](struct.objects.AutoArray) struct.
     pub fn get_array_elements<T: TypeArray>(
-        &self,
+        self,
```

Comment:
I would leave all these as they are. Even though the `JNIEnv` wrapper is minimal, using references is better.

And, handle all the changes just inside `AutoArray`.

Reply:
I had made this change because passing by reference means extra pointer chasing, which is a tiny bit slower everywhere. But it *is* everywhere, since all the other APIs still work by reference, so I guess it makes sense to change that all in one go (and for now to leave it as is).

Reply:
Yes. In any case, let's create an issue to change all those references at once.

Measuring / benchmarking wouldn't be bad either, to confirm which option is faster